sphinx/writers/latex.py

```diff
@@ -2163,26 +2147,36 @@ def depart_title_reference(self, node):
         # type: (nodes.Node) -> None
         self.body.append('}')
 
+    def visit_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        longest_label = max((subnode[0].astext() for subnode in node), key=len)
+        self.body.append(u'\n\\begin{sphinxthebibliography}{%s}\n' % longest_label)
+
+    def depart_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        self.body.append(u'\\end{sphinxthebibliography}\n')
+
     def visit_citation(self, node):
         # type: (nodes.Node) -> None
-        # TODO maybe use cite bibitems
-        # bibitem: [citelabel, citetext, docname, citeid]
-        self.bibitems.append(['', '', '', ''])
-        self.context.append(len(self.body))
+        label = node[0].astext()
+        self.body.append(u'\\bibitem[%s]{%s:%s}{' %
+                         (self.encode(label), node['docname'], node['ids'][0]))
 
     def depart_citation(self, node):
         # type: (nodes.Node) -> None
-        size = self.context.pop()
-        text = ''.join(self.body[size:])
-        del self.body[size:]
-        self.bibitems[-1][1] = text
+        self.body.append('}\n')
 
     def visit_citation_reference(self, node):
         # type: (nodes.Node) -> None
-        # This is currently never encountered, since citation_reference nodes
-        # are already replaced by pending_xref nodes in the environment.
-        self.body.append('\\cite{%s}' % self.idescape(node.astext()))
-        raise nodes.SkipNode
+        if self.in_title:
+            pass
+        else:
+            self.body.append('\\cite{%s:%s}' % (node['docname'], node['refname']))
```

Comment:
Would it be very complicated to achieve here if possible 

    '\\sphinxcite{%s:%s}{%s}' % (node['docname'], node['refname'], actual citation text)

Then we define 

    \protected\def\sphinxcite #1#2{\cite{#1}}

but we have options to do other things in future, like printing the citations at location. Or people can possibly use ``\citep`` or ``citet`` from `natbib`  package (not checked if that make sense with thebibliography).

At any rate, I suggest `\sphinxcite` not `\cite`.

Reply:
actually, forget that. I am not sure exactly what I wanted to achieve. But do keep the `\sphinxcite` with `\newcommand\sphinxcite{\cite}` in sphinx.sty. Perhaps with `\protected\def\sphinxcite{\cite}` we can even use `\cite` in titles (but that may not be compatible with user loading extra packages modifying behaviour of `\cite`) I need to test.

Reply:
done

Reply:
Note: On my local, `\sphinxcite` does not work even if `\protected`. So I keep the `visit_citation_reference` as is.